server.go

```diff
@@ -338,12 +336,60 @@ Loop:
 		}
 	}
 
-	s.log.Println("Waiting for in-flight connections...")
+	wg.Wait()
 	inflightWg.Wait()
-	s.log.Println("Bye")
+	managerWg.Wait()
 	return nil
 }
 
+// shutdown closes current clients and also stops accepting new clients in a
+// non-blocking manner
+func (s *Server) shutdown() {
+	// stop accepting new consumers
+	s.managerCancel()
```

Comment:
I think this should be a blocking call. What about something like `s.manager.StopAcceptingConsumers()` to push this behavior to the `ConsumerManager`?

Reply:
Done.